src/engine/BMTournament.php

```diff
@@ -0,0 +1,688 @@
+ * @property-read int    $roundNumber            Current round number of tournament
+ * @property      int    $gameMaxWins            Max wins per game in this tournament round
+ * @property-read array  $playerIdArray          Array of player IDs taking part in the tournament
+ * @property-read array  $remainCountArray       Array of remain chances for each player in the tournament
+ * @property-read array  $buttonIdArrayArray     Array of arrays of button IDs, indexed by player ID
+ * @property-read array  $gameIdArrayArray       Array of arrays of game IDs, zero-indexed
+ * @property-read BMTournamentState $tournamentState Current tournament state as an enum
+ *
+ * @SuppressWarnings(PMD.TooManyFields)
+ */
+abstract class BMTournament {
+    // properties -- all accessible, but written as protected to enable the use of
+    //               getters and setters
+
+    /**
+     * Tournament ID in the database
+     *
+     * @var int
+     */
+    protected $tournamentId;
+
+    /**
+     * Type of tournament
+     *
+     * @var string
+     */
+    protected $type;
+
+    /**
+     * Player ID of tournament creator
+     *
+     * @var int
+     */
+    protected $creatorId;
+
+    /**
+     * Name of tournament
+     *
+     * @var string
+     */
+    protected $name;
+
+    /**
+     * Description of tournament
+     *
+     * @var string
+     */
+    protected $description;
+
+    /**
+     * Number of players
+     *
+     * @var int
+     */
+    protected $nPlayers;
+
+    /**
+     * Number of days before a game is auto-advanced
+     *
+     * @var int
+     */
+    protected $autoAdvanceInterval;
+
+    /**
+     * Do we hide players' names initially?
+     *
+     * @var bool
+     */
+    protected $arePlayersAnonymous;
+
+    /**
+     * Do we initially shuffle the order of players?
+     *
+     * @var bool
+     */
+    protected $doShufflePlayers;
+
+    /**
+     * Do we hide the selected button names initially?
+     *
+     * @var bool
+     */
+    protected $areButtonsAnonymous;
+
+    /**
+     * Current round number of the tournament
+     *
+     * @var int
+     */
+    protected $roundNumber;
+
+    /**
+     * Maximum number of wins for each game in this round of the tournament
+     *
+     * @var int
+     */
+    protected $gameMaxWins;
+
+    /**
+     * Array of player IDs taking part in the tournament
+     *
+     * @var array
+     */
+    protected $playerIdArray;
+
+    /**
+     * Array of remain chances for each player in the tournament
+     *
+     * @var type
+     */
+    protected $remainCountArray;
+
+    /**
+     * Array of arrays of button IDs, indexed by player ID
+     *
+     * @var array
+     */
+    protected $buttonIdArrayArray;
+
+    /**
+     * Array of arrays of game IDs that are part of the tournament
+     *
+     * The outer array is indexed by round number (zero-indexed),
+     * the inner array by position number (zero-indexed)
```

Comment:
Position number is stored in tournament_player_map. It is significant, because it determines the matchups that occur.

Shuffling position number happens in a dedicated phase after everyone has joined.